add solana RPC methods (#1681)
WalletConnect/walletconnect-docs: docs/advanced/multichain/rpc-reference/solana-rpc.md

---
description: Solana JSON-RPC Methods & Events
---

# Solana

The Chain IDs of Solana can be found in [namespaces.chainagnostic.org](https://namespaces.chainagnostic.org/solana/caip10) [(GitHub)](https://github.com/ChainAgnostic/namespaces/blob/main/solana/caip10.md).

## Notifications (Events)

### accountChanged

> This methods is sent by the wallet when the active _account_ changes.

Reviewer: ```suggestion
This RPC notification is sent by the wallet when the active _account_ changes in the wallet UX.
```

}
```

### solana_signMessage

This method returns a signature for the provided message from the requested signer address.

**Parameters**

    1. `Object` - Signing parameters:
    	1.1. `message` : `String` -  the message to be signed (base58 encoded)
    	1.2. `pubkey` : `String` -  public key of the signer

**Returns**

    1. `Object`
    	1.1. `signature` : `String` - corresponding signature for signed message

**Example**

```javascript
// Request
{
	"id": 1,
	"jsonrpc": "2.0",
	"method": "solana_signMessage",
	"params": {
		"message": "37u9WtQpcm6ULa3VtWDFAWoQc1hUvybPrA3dtx99tgHvvcE7pKRZjuGmn7VX2tC3JmYDYGG7",
		"pubkey": "AqP3MyNwDP4L1GJKYhzmaAUdrjzpqJUZjahM7kHpgavm"
	}
}

// Result
{
	"id": 1,
	"jsonrpc": "2.0",
	"result":  { signature: "2Lb1KQHWfbV3pWMqXZveFWqneSyhH95YsgCENRWnArSkLydjN1M42oB82zSd6BBdGkM9pE6sQLQf1gyBh8KWM2c4" }
}
```

### solana_signTransaction

This method returns a signature over the provided instructions by the targeted public key.

**Parameters**

    1. `Object` - Signing parameters:
    	1.1. `transaction` : `TODO` -  the transaction raw bytes encoded on base58
    	1.2. `pubkey` : `TODO` - TODO

> **Returns**

Reviewer: returns a promise for a signature

	"jsonrpc": "2.0",
	"method": "solana_signTransaction",
	"params": {

>       "transaction": "abc123...",

Author: ```suggestion
      "transaction": "vxBNpvao9QJmLKXUThbbjRnxm3ufu4Wku97kHd5a67FDjSqeHwcPrBKTjAHp4ECr61eWwoxvUEVTuuWX65P9bCNDJrTJpWyhiQ68iePVXQba65aCqANmtsokLnysL7b7h3JaL7u1Wk93u6NJ6Ff1gjFy3aPBtoBBQLXPcUS5xmriZA21673DYxgUsf9Efq25bt3Doqg8qNAzQdHs9adfeQoUDq1YriVhKrxPYstfwANvrpA6bfoMAbzw1Ssix3hesqL9sSFM6WnbU1KPqt4i8dE6s1cj64wDgGJmjzX",
```

      "pubkey": "AqP3MyNwDP4L1GJKYhzmaAUdrjzpqJUZjahM7kHpgavm"
	}
}

// Result
{
	"id": 1,
	"jsonrpc": "2.0",
	"result":  { 
      "transaction": "abc123..."
    }
}
```

### solana_signAndSendTransaction

This method request to a wallet to sign and submit a transaction to the network.

**Parameters**

    1. `Object` - Signing parameters:
    	1.1. `transaction` : `TODO` -  the transaction raw bytes encoded on base58
    	1.2. `pubkey` : `TODO` - TODO
			1.3. `sendOptions` : `object` - TODO
				1.3.1. `maxRetries` : `number` \(optional\) - 
				1.3.1. `minContextSlot` : `number` \(optional\) -
				1.3.1. `preflightCommitment` : `string` \(optional\) -
				1.3.1. `skipPreflight` : `boolean` \(optional\) -

**Returns**

    1. `Object`
    	1.1. `transaction` : `TODO` - the signed transaction raw bytes encoded on base58

**Example**

```javascript
// Request
{
	"id": 1,
	"jsonrpc": "2.0",
	"method": "solana_signAndSendTransaction",
	"params": {

>       "transaction": "abc123...",

Author: ```suggestion
      "transaction": "vxBNpvao9QJmLKXUThbbjRnxm3ufu4Wku97kHd5a67FDjSqeHwcPrBKTjAHp4ECr61eWwoxvUEVTuuWX65P9bCNDJrTJpWyhiQ68iePVXQba65aCqANmtsokLnysL7b7h3JaL7u1Wk93u6NJ6Ff1gjFy3aPBtoBBQLXPcUS5xmriZA21673DYxgUsf9Efq25bt3Doqg8qNAzQdHs9adfeQoUDq1YriVhKrxPYstfwANvrpA6bfoMAbzw1Ssix3hesqL9sSFM6WnbU1KPqt4i8dE6s1cj64wDgGJmjzX",
```

      "pubkey": "AqP3MyNwDP4L1GJKYhzmaAUdrjzpqJUZjahM7kHpgavm",
			"sendOptions": {
				"maxRetries": 2;
				"minContextSlot": TODO;
				"preflightCommitment": "processed";
				"skipPreflight": false;
			}
	}
}

// Result
{
	"id": 1,
	"jsonrpc": "2.0",
	"result":  { 

>       "transaction": "abc123..."

Author: ```suggestion
      "transaction": "..."
```
